src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -453,7 +453,8 @@ private String DecodeEntryName(Byte[] entryNameBytes)
                 readEntryNameEncoding = Encoding.UTF8;
             }
 
-            return new string(readEntryNameEncoding.GetChars(entryNameBytes)); //readEntryNameEncoding.GetString(entryNameBytes);
+            fixed (byte* bytePtr = entryNameBytes)
+                return readEntryNameEncoding.GetString(bytePtr, entryNameBytes.Length);
```

Comment:
Because for some bizarre reason the default implementation of `GetString(byte[])` calls the exact same code I've just removed, while `GetString(byte*, int)` does not (refer [here](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Text/Encoding.cs#L1492-L1496) and [here](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Text/Encoding.cs#L1322-L1336)). I kept the change in my earlier PR #2691 because ASCIIEncoding overrides this, but here we don't know the type of the encoding beforehand.